### ISSUE TITLE
sun.misc.Unsafe deprecation in Java 24

### ISSUE THREAD

**Issue by eygraber** [pinterest/ktlint#2973]
After updating to Java 24 I get the following warning printed:

```
WARNING: A terminally deprecated method in sun.misc.Unsafe has been called
WARNING: sun.misc.Unsafe::objectFieldOffset has been called by org.jetbrains.kotlin.com.intellij.util.containers.Unsafe (file:/home/eli/workspace/my-project/tmp/ktlint-1.5.0)
WARNING: Please consider reporting this to the maintainers of class org.jetbrains.kotlin.com.intellij.util.containers.Unsafe
WARNING: sun.misc.Unsafe::objectFieldOffset will be removed in a future release
```

[An issue has already been filed with Intellij](https://youtrack.jetbrains.com/issue/IDEA-370928), so I'm not sure if there's anything for ktlint to do but track it, but I figured I'd file it here just in case.

**Comment by paul-dingemans** [comment]
Tnx for reporting

> [An issue has already been filed with Intellij](https://youtrack.jetbrains.com/issue/IDEA-370928), so I'm not sure if there's anything for ktlint to do but track it, but I figured I'd file it here just in case.

Indeed, there is nothing that ktlint can do about this as we need the embedded kotlin compiler.

**Comment by paul-dingemans** [comment]
As per https://openjdk.org/jeps/498 ktlint cli can suppress this warning by setting `--sun-misc-unsafe-memory-access=allow` on line https://github.com/pinterest/ktlint/blob/73c92fa05e5d9dbf8f00a3c8f6ba2799352fac0d/ktlint-cli/build.gradle.kts#L100

**Comment by paul-dingemans** [comment]
Closed by #3040 and #3042
